aquadoggo/src/network/service.rs

```diff
@@ -140,13 +110,34 @@ pub async fn network_service(
                 } => {
                     debug!("ConnectionClosed: {peer_id} {endpoint:?} {num_established} {cause:?}")
                 }
+                SwarmEvent::ConnectionEstablished { peer_id, .. }
+                    if network_config.rendezvous_client
+                        && network_config.rendezvous_peer_id.unwrap() == peer_id =>
+                {
+                    debug!(
+                        "Connected to rendezvous point, discovering nodes in '{NODE_NAMESPACE}' namespace ..."
+                    );
+
+                    swarm
+                        .behaviour_mut()
+                        .rendezvous_client
+                        .as_mut()
```

Comment:
This looks messy, unfortunately. We have to call `.as_mut()` because the `rendezvous_client` behaviour is behind a `Toggle` type.